### ISSUE TITLE
It's not clear how to run the project in VirtualBox

### ISSUE THREAD

**Issue by rexim** [tsoding/pinpog#38]

**Comment by elenderg** [comment]
You should run the game in QEMU

**Comment by Vyjill** [comment]
# How to run this project on VirtualBox ?

TL;DR : use qemu.

## Create a new virtual machine

- Type : Other
- Version : Other/Unknown
![image](https://github.com/tsoding/pinpog/assets/69155487/d4d03cea-765b-4492-b600-5e976c6c305b)

- Ram : 4M
- Proc : 1 CPU
![image](https://github.com/tsoding/pinpog/assets/69155487/f86d2390-37cb-446a-9312-28aa0efb48dd)

- Do not add a virtual disk
![image](https://github.com/tsoding/pinpog/assets/69155487/1fc8c2ea-2507-4d24-bcc5-8d62d26ff527)


## Create a floppy disk containing the game

Assuming you already compiled the game, you should have the `pinpog` binary in the project directory.

### Using windows powershell

```powershell
$pinpog = [System.IO.File]::ReadAllBytes('pinpog')
$floppy = $pinpog + @(0) * (1474560-$bytes.Length)
[io.file]::WriteAllBytes('floppy_pinpog.vfd', $towrite)
```

### Using linux bash

```bash
head -c 1474560 /dev/zero > floppy_pinpog.vfd
sudo dd if=./pinpog of=./floppy_pinpog.vfd
```

### Using python

```py
pinpog_file = open("pinpog", "rb")
floppy_file = open("floppy_pinpog.vfd", "wb")

game_bytes = pinpog_file.read()

floppy_file.write(game_bytes + b'\x00'*(1474560-len(game_bytes)))

pinpog_file.close()
floppy_file.close()
```

## Add the floppy disk to Virtualbox storage

In your virtual machine configuration > storage, add a floppy controller `I82078 (floppy)`. Then add the newly created floppy image to the floppy controller (Add > Add > select the `.vfd` file).

![image](https://github.com/tsoding/pinpog/assets/69155487/a0170bc2-2be0-439c-9008-a64f7c6da2ae)
![image](https://github.com/tsoding/pinpog/assets/69155487/1cdf054a-800b-470a-a6b0-8c9c23223c06)

---

You can now run pinpog on VirtualBox.
